### PR TITLE
Cleanup in StageBundleFactory, ProcessBundleDescriptors

### PR DESCRIPTION
**Pull request by tgroh**
DESCRIPTION HERE

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
R: @bsidhom

File: runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java [hunk 1]
```diff
@@ -51,8 +51,8 @@
 // TODO: Rename to ExecutableStages?
 public class ProcessBundleDescriptors {
   public static ExecutableProcessBundleDescriptor fromExecutableStage(
-      String id, ExecutableStage stage, Components components, ApiServiceDescriptor dataEndpoint)
-      throws IOException {
+      String id, ExecutableStage stage, ApiServiceDescriptor dataEndpoint) throws IOException {
+    Components components = stage.getComponents();
```
Comment (bsidhom) [inline runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java]: Good catch.

File: runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java [hunk 2]
```diff
@@ -190,12 +190,13 @@ public static ExecutableProcessBundleDescriptor of(
...
      */
-    public abstract RemoteInputDestination<WindowedValue<?>> getRemoteInputDestination();
+    public abstract RemoteInputDestination<? super WindowedValue<?>> getRemoteInputDestination();
 
     /**
      * Get all of the targets materialized by this {@link ExecutableProcessBundleDescriptor} and the
      * java {@link Coder} for the wire format of that {@link BeamFnApi.Target}.
      */
-    public abstract Map<BeamFnApi.Target, Coder<WindowedValue<?>>> getOutputTargetCoders();
+    public abstract Map<BeamFnApi.Target, Coder<? extends WindowedValue<?>>>
```
Comment (bsidhom) [inline runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java]: I'm not sure we want to modify the coder type here.

Comment (tgroh) [reply runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java]: Done.

The `? super` means we don't have to do the double cast between `(Coder<WindowedValue<T>>) (Coder)`, which is why I'm keeping that one.

Comment (bsidhom) [reply runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java]: Yes, I think we should keep the change for the sink-only data receiver.

**Comment by tweise** [comment]
The pre-commit build failed on other tests, before it even got to the error introduced here. We should probably only merge on green.

**Pull request merged**